nims-notes-things/src/components/Form.js

```diff
@@ -58,55 +51,60 @@ const Form = (props) => {
         ...formData,
         id: uniqueId,
       };
-      const notesCopy = [...notes];
-      notesCopy.push(formDataCopy);
-      dispatch(setNote(notesCopy));
+      dispatch(setNote(formDataCopy));
+      alert("Note has been created :) ");
     }
   };
 
-  const handleOnBlur = (value, key) => {
-    setFormEmpty((previousState) => {
-      return {
-        ...previousState,
-        [key]: value === "",
-      };
-    });
+  const handleOnBlur = (key) => {
+    emptyData.current[key].innerHTML =
+      formData[key] === "" ? `${key} required` : "";
   };
 
   return (
     <StyledForm onSubmit={handleOnSubmit}>
       <StyledDiv>
         <InnerForm>
-          {empty.name && <ErrorMessage>Name field is required</ErrorMessage>}
+          <ErrorMessage
+            id="name"
+            ref={(element) => (emptyData.current["name"] = element)}
+          ></ErrorMessage>
           <Input
             name="name"
             placeholder="Please enter Your name"
             onChange={(e) => handleOnChange(e, "name")}
-            onBlur={(e) => handleOnBlur(e.target.value, "name")}
+            onBlur={(e) => handleOnBlur("name")}
             value={formData.name}
           />
-          {empty.date && <ErrorMessage>Date field is required</ErrorMessage>}
+          <ErrorMessage
+            id="date"
+            ref={(element) => (emptyData.current["date"] = element)}
+          ></ErrorMessage>
           <Input
             name="date"
             type="date"
             placeholder="Please enter the date"
             onChange={(e) => handleOnChange(e, "date")}
-            onBlur={(e) => handleOnBlur(e.target.value, "date")}
+            onBlur={(e) => handleOnBlur("date")}
             value={formData.date}
           />
-          {empty.note && <ErrorMessage>Note field is required</ErrorMessage>}
+          <ErrorMessage
+            id="note"
+            ref={(element) => (emptyData.current["note"] = element)}
+          ></ErrorMessage>
           <TextArea
             name="note"
             placeholder="Please enter Your Note"
             onChange={(e) => handleOnChange(e, "note")}
-            onBlur={(e) => handleOnBlur(e.target.value, "note")}
+            onBlur={(e) => handleOnBlur("note")}
             value={formData.note}
           />
         </InnerForm>
       </StyledDiv>
-      <Button background="red" type="submit">
+      <Button background="blue" type="submit" to="/notes">
         Submit
       </Button>
+      {notes.length === 0 && <p> There are no notes currently</p>}
```

Comment:
This should probably move to the notes page. As it would make sense to let people know that there aren't any notes created there.